kahuna/public/js/search/index.js

```diff
@@ -35,15 +35,28 @@ export var search = angular.module('kahuna.search', [
 search.config(['$stateProvider', '$urlMatcherFactoryProvider',
                function($stateProvider, $urlMatcherFactoryProvider) {
 
-    // Query params slashes are encoded that mess up our seach query when we search for e.g. `24/7`.
-    // see: https://github.com/angular-ui/ui-router/issues/1119#issuecomment-64696060
     // TODO: Fix this
-    $urlMatcherFactoryProvider.type('Slashed', {
-        encode: val => angular.isDefined(val) ? val : undefined,
-        decode: val => angular.isDefined(val) ? val : undefined,
-        // We want to always match this type. If we match on slash, it won't update
-        // the `stateParams`.
-        is: () => true
+    $urlMatcherFactoryProvider.type('Query', {}, () => {
+        var queryFilters = {
+            // Query params slashes are encoded that mess up our seach query when we search for
+            // e.g. `24/7`.
+            // see: https://github.com/angular-ui/ui-router/issues/1119#issuecomment-64696060
+            defined: val => angular.isDefined(val) ? val : undefined,
+            // The string replacement is for removing the invisible "zero-width-space" UTF-8
+            // causing queries to fail: https://en.wikipedia.org/wiki/Zero-width_space
+            removeUtf8SpecialChars: val => val.replace(/%E2%80%8B/ig, '%20')
+        };
+        queryFilters.call = (val) => {
```

Comment:
we could just have this as a call function (to avoid mutating the obj). Not fussed, just a thought.